.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,39 @@
+> [!NOTE]
+> In the Title field above, Provide a succinct and descriptive title for the pull request, e.g., "Improve caching mechanism for API calls"
+
+## Description
+
+[Provide a detailed explanation of the changes you have made. Include the reasons behind these changes and any relevant context. Link any related issues.]
+
+## Type of Change
+
+_Put an `x` in all boxes that apply_
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] Hot fix (corrects a major software bug or fault and should released as quickly as possible.)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] Documentation update
+- [ ] Refactoring
+- [ ] Security patch
+- [ ] UI/UX improvement
+
+## Testing
+
+[Detail the testing you have performed to ensure that these changes function as intended. Include information about any added tests.]
+
+## Impact
+
+[Discuss the impact of your changes on the project. This might include effects on performance, new dependencies, or changes in behaviour.]
+
+## Additional Information
+
+[Any additional information that reviewers should be aware of.]
```

Comment:
Maybe add a remark about images when implementing visual changes?

Reply:
Yup, that should be there. Something is like A picture is required for a when implementing a visual change or a before and after picture if applicable.